packages/react-native/Libraries/Text/TextInput/Multiline/RCTUITextView.mm

```diff
@@ -307,11 +334,44 @@ - (void)_updatePlaceholder
 
 - (CGRect)caretRectForPosition:(UITextPosition *)position
 {
+  CGRect originalRect = [super caretRectForPosition:position];
+
   if (_caretHidden) {
     return CGRectZero;
   }
 
-  return [super caretRectForPosition:position];
+  if(_caretYOffset != 0) {
+    originalRect.origin.y += _caretYOffset;
+  }
+
+  if(_caretHeight != 0) {
+    originalRect.size.height = _caretHeight;
+  }
+
+  return originalRect;
+}
+
+- (NSArray *)selectionRectsForRange:(UITextRange *)range {
+    NSArray *superRects = [super selectionRectsForRange:range];
+    if(_caretYOffset != 0 && _caretHeight != 0) {
+        NSMutableArray *customTextSelectionRects = [NSMutableArray array];
+
+        for (UITextSelectionRect *rect in superRects) {
+            CustomTextSelectionRect *customTextRect = [[CustomTextSelectionRect alloc] init];
+
+            customTextRect.rect = CGRectMake(rect.rect.origin.x, rect.rect.origin.y + _caretYOffset, rect.rect.size.width, _caretHeight);
```

Comment:
This then needs to handle `_caretHeight` being potentially 0.

Reply:
The block should only be executed if both properties are set.